docs/html/reference/pip_install.rst

```diff
@@ -1123,37 +1123,45 @@ Examples
 
          py -m pip install --index-url http://my.package.repo/simple/ SomePackage
 
-   Search an additional index during install, in addition to `PyPI`_
+   Install from a local flat directory containing archives (and don't scan indexes):
 
    .. tab:: Unix/macOS
 
       .. code-block:: shell
 
-         python -m pip install --extra-index-url http://my.package.repo/simple SomePackage
+         python -m pip install --no-index --find-links=file:///local/dir/ SomePackage
+         python -m pip install --no-index --find-links=/local/dir/ SomePackage
+         python -m pip install --no-index --find-links=relative/dir/ SomePackage
 
    .. tab:: Windows
 
       .. code-block:: shell
 
-         py -m pip install --extra-index-url http://my.package.repo/simple SomePackage
+         py -m pip install --no-index --find-links=file:///local/dir/ SomePackage
+         py -m pip install --no-index --find-links=/local/dir/ SomePackage
+         py -m pip install --no-index --find-links=relative/dir/ SomePackage
 
-   Install from a local flat directory containing archives (and don't scan indexes):
+   Search an additional index during install, in addition to `PyPI`_
+
+   .. warning::
+
+       Using this option to search for packages which are not in the main
+       repository (such as private packages) is unsafe, per a security
+       vulnerability called "dependency confusion": an attacker can claim the
+       package on the public repository in a way that will ensure it gets
+       chosen over the private package.
```

Comment:
Instead of just drop the name here, I would add a reference to explain what this is. Since the original research writeup is on Medium (and may be paywalled), it’s probably reasonable to link to Microsoft’s whitepaper on this topic. The whitepaper also provides solutions to the issue, which is nice.

https://azure.microsoft.com/en-us/resources/3-ways-to-mitigate-risk-using-private-package-feeds/

Feel free to provide better reference material if you know one!

I also wonder if it’s a good idea to use “dependency confusion” here, since the name is not agreed within the community (and is arguably, uh, confusing 😛). The Microsoft whitepaper describes this as “hybrid configuration”, for example.

Reply:
Oh what a time to be alive when Microsoft is the more open option... I do agree with that suggestion though.

Regarding the naming: “hybrid configuration” doesn't result in much relating to this particular problem when I plug it into a search engine. In a newer [white paper](https://azure.microsoft.com/en-us/resources/3-ways-to-mitigate-risk-using-private-package-feeds/) this is referred to as a "substitution attack", but that has the problem of being overloaded with "algorithm substitution attacks" when searching. When specifically plugging in "dependency substitution attack", the top results all mention "dependency confusion" in their titles, so I'm tempted to just stick with that.

Reply:
That makes perfect sense, let’s do it.